crates/kitsune_p2p/proxy/src/bin/proxy-stress.rs

```diff
@@ -0,0 +1,240 @@
+use futures::{sink::SinkExt, stream::StreamExt};
+use ghost_actor::dependencies::tracing;
+use kitsune_p2p_proxy::*;
+use kitsune_p2p_transport_quic::*;
+use kitsune_p2p_types::{
+    dependencies::{ghost_actor, url2},
+    transport::*,
+    transport_mem::*,
+};
+use structopt::StructOpt;
+
+/// Proxy transport selector
+#[derive(structopt::StructOpt, Debug, Clone, Copy)]
+pub enum ProxyTransport {
+    /// Use the local in-process memory transport (faster, uses more memory)
+    Mem,
+
+    /// Use the real QUIC/UDP transport (slower, more realistic)
+    Quic,
+}
+
+const E: &str = "please specify 'mem'/'m' or 'quic'/'q'";
+impl std::str::FromStr for ProxyTransport {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.chars().next() {
+            Some('M') | Some('m') => Ok(Self::Mem),
+            Some('Q') | Some('q') => Ok(Self::Quic),
+            _ => Err(E),
+        }
+    }
+}
+
+/// Option Parsing
+#[derive(structopt::StructOpt, Debug, Clone)]
+#[structopt(name = "proxy-stress")]
+pub struct Opt {
+    /// Transport to test with. ('mem'/'m' or 'quic'/'q')
+    #[structopt(short = "t", long, default_value = "Mem")]
+    transport: ProxyTransport,
+
+    /// How many client nodes should be spawned
+    #[structopt(short = "n", long, default_value = "128")]
+    node_count: u32,
+
+    /// Interval between requests per node
+    #[structopt(short = "i", long, default_value = "200")]
+    request_interval_ms: u32,
+
+    /// How long nodes should delay before responding
+    #[structopt(short = "d", long, default_value = "200")]
+    process_delay_ms: u32,
+}
+
+#[tokio::main]
+async fn main() {
+    let _ = ghost_actor::dependencies::tracing::subscriber::set_global_default(
+        tracing_subscriber::FmtSubscriber::builder()
+            .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+            .finish(),
+    );
```

Comment:
Observability is already a dependency, if you use that to start tracing then you can get access to cross process traces if that's important.